### PR TITLE
Add docs for generate-pipelines command

### PR DESCRIPTION
**Pull request by stealthycoin**
cc @jamesls @kyleknap

### CODE REVIEW

File: docs/source/topics/cd.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
+Continuous Deployment (CD)
+===========================
+
+Chalice can be used to set up a baisc Continuous Deployment pipeline.The
```
Comment (jamesls) [inline docs/source/topics/cd.rst]: s/baisc/basic/

File: docs/source/topics/cd.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+``chalice deploy`` command is good for getting up and running quickly with
+Chalice, but in a team environment properly managing permissions and sharing
+and updating the ``deployed.json`` file will get messy.
+
+One way to scale up your chalice app is to create a continuous deployment
+pipeline. The pipeline can run tests on code changesand, if they pass, promote
```
Comment (jamesls) [inline docs/source/topics/cd.rst]: s/changesand/changes and/

File: docs/source/topics/cd.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+several new AWS resources that make up a bare bones CD pipeline.
+
+* **CodeCommit Repository** - The `CodeCommit <https://aws.amazon.com/codecommit/>`_
+  repository is the entrypoint into the pipeline. Any code you want to deploy
+  should be pushed to this remote.
+* **CodePipeline Pipeline** - The
+  `CodePipeline <https://aws.amazon.com/codepipeline/>`_ is what coordinates
+  the build process, and pushes the released code out.
+* **CodeBuild Project** - The `CodeBuild <https://aws.amazon.com/codebuild/>`_
+  project is where the code bundle is built that will be pushed to lambda. The
```
Comment (jamesls) [inline docs/source/topics/cd.rst]: s/lambda/Lambda

File: docs/source/topics/cd.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+packaging your chalice app, and a CodePipeline stage to deploy your
+application using CloudFormation.
+
+
+Usage example
+-------
+
+Setting up the deployment pipeline is a two step process. First use the
+``chalice generate-pipeline`` command to generate a base CloudFormation
+template. Second use the AWS CLI to deploy the cloudformation template using
```
Comment (kyleknap) [inline docs/source/topics/cd.rst]: Wrong casing on cloudformation.

File: docs/source/topics/cd.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+`here <http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html>`_
+.
+
+
+CodePipeline
+------------
+
+CodePipeline is the main coordinator between all the other resources. It
+watches for changes on the CodeCommit repository, and triggers builds in the
+CodeBuid project. If the build succeeds then it will start a CloudFormation
```
Comment (kyleknap) [inline docs/source/topics/cd.rst]: CodeBuid --> CodeBuild

File: docs/source/topics/cd.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+makes it easy to kick off deployments. The developer doing the deployment only
+needs to push the code to release up to the CodeCommit repository master branch
+All the developer needs is keys that allow for push access to the CodeCommit
+repository. This is a lot easier than managing a set of ``deployed.json``
+resources across a repsoitory and manually doing ``chalice deploy`` whenever
+a change needs to be deployed.
+
+The default CodeCommit repository that is created is empty, you will have to
+populate it with the Chalice application code. Permissions will also need to be
+setup, you can find the documentation on how to do that
```
Comment (JordonPhillips) [inline docs/source/topics/cd.rst]: setup as a single word is a noun

File: docs/source/topics/cd.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+
+CodePipeline
+------------
+
+CodePipeline is the main coordinator between all the other resources. It
+watches for changes on the CodeCommit repository, and triggers builds in the
+CodeBuild project. If the build succeeds then it will start a CloudFormation
+deployment of the built artifacts to a beta stage. This should be treated as
+a starting point not a fully featured CD system.
```
Comment (JordonPhillips) [inline docs/source/topics/cd.rst]: comma after starting point

File: docs/source/topics/cd.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+          tmp/packaged/sam.json --s3-bucket ${APP_S3_BUCKET}
+          --output-template-file transformed.yaml
+  artifacts:
+    type: zip
+    files:
+      - transformed.yaml
+
+The CodeBuild stage installs both the AWS CLI and Chalice, then creates a
+package out of your chalice project, pushing the package to the application
+S3 bucket that was created for you. The transformed cloudformation template
```
Comment (JordonPhillips) [inline docs/source/topics/cd.rst]: CloudFormation

File: docs/source/topics/cd.rst [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+is the only artifact, and can be run by CodePipeline after the build has
+succeeded.
+
+
+Deploying to beta stage
+-----------------------
+
+Once the CodeBuild stage has finished building the Chalice package and
+creating the ``transformed.yaml``, CodePipeline will take these artifacts and
+build use them to create or update the beta stage. The ``transformed.yaml``
```
Comment (JordonPhillips) [inline docs/source/topics/cd.rst]: -build

**Pull request merged**